### PR TITLE
JNG-4216 Wrap SDK UUID as an Identifier interface

### PR DESCRIPTION
**Pull request by robertcsakany**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4216" title="JNG-4216" target="_blank">Sub-taskJNG-4216</a>  Wrap UUID identifier as Identifiable and using that for resolve. 
  </td></table>
  <br />
 

JNG-4216 Wrap SDK UUID as an Identifier interface

### CODE REVIEW

**Review by bencegelei — APPROVED** [review]

File: README.adoc [hunk 1]
```diff
@@ -257,6 +260,80 @@ User user = MapBuilderProxy.builder(UserBuilder.class, User.class).newInstance()
 UserBean bean = user.adaptTo(UserBean.class);
 ----
 
+=== hashCode
+You cen define `hashCode` method in interface. On that case that method will be
```
Comment (gaborflorian) [inline README.adoc]: can

File: README.adoc [hunk 2]
```diff
@@ -257,6 +260,80 @@ User user = MapBuilderProxy.builder(UserBuilder.class, User.class).newInstance()
...
+                Objects.toString(o1.getId(), "null"),
+                Objects.toString(o1.getNote(), "null")
+        );
+    }
+}
+----
+
+=== equals
+
+You cen define `equals` method in interface. On that case that method will be
```
Comment (gaborflorian) [inline README.adoc]: can Victim of the copy-paste

File: README.adoc [hunk 3]
```diff
@@ -257,6 +260,80 @@ User user = MapBuilderProxy.builder(UserBuilder.class, User.class).newInstance()
...
+   static int hashCode(A o) {
+      o.getFld.hashCode()
+   }
+}
+----
+
+
+=== toString
+
+You cen define `toString` method in interface. On that case that method will be
```
Comment (gaborflorian) [inline README.adoc]: can Victim of the copy-paste

File: README.adoc (hunk 1 above)
Comment (noherczeg) [inline README.adoc]: "You can define a static `hashCode` method on the interface"

File: README.adoc (hunk 3 above)
Comment (noherczeg) [inline README.adoc]: "You can define a static `toString` method on the interface"

File: README.adoc [hunk 4]
```diff
@@ -257,6 +260,80 @@ User user = MapBuilderProxy.builder(UserBuilder.class, User.class).newInstance()
...
+performed on the proxy's object call. If not defined the fields ara iterated and printed.
```
Comment (noherczeg) [inline README.adoc]: are

File: README.adoc (hunk 2 above)
Comment (noherczeg) [inline README.adoc]: "You can define a static equals method on the interface"

File: src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java [hunk 5]
```diff
@@ -220,6 +266,12 @@ public Map<String, AttributeInfo> load(Class sourceClass) throws Exception {
             .expireAfterAccess(Long.parseLong(System.getProperty("structuredMapProxyCacheExpireInSecond", "60")), TimeUnit.SECONDS)
             .build(typeInfoCacheLoader);
 
+    private static LoadingCache<Class, Map<String, Method>> staticMethodCache = CacheBuilder
+            .newBuilder()
+            .expireAfterAccess(Long.parseLong(System.getProperty("structuredMapProxyCacheExpireInSecond", "60")), TimeUnit.SECONDS)
```
Comment (noherczeg) [inline src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java]: Undocumented system property.

**Review by noherczeg — CHANGES_REQUESTED** [review]

**Review by noherczeg — APPROVED** [review]

**Pull request merged**